src/EffectiveCSharp.Analyzers/PreferDeclarationInitializersToAssignmentStatementsCodeFixProvider.cs

```diff
@@ -0,0 +1,257 @@
+                            equivalenceKey: "Use member initializer"),
+                        diagnostic);
+                    break;
+                default:
+                    continue;
+            }
+        }
+
+        foreach (IFieldSymbol fieldSymbol in initializationsToFixPerField.Keys)
+        {
+            if (initializationsToFixPerField.TryGetValue(fieldSymbol, out IList<ExpressionStatementSyntax> initializations))
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: "Initialize in field declaration instead",
+                        createChangedSolution: cancellationToken => ReplaceAssignmentsWithFieldDeclarationInitializerAsync(context.Document, fieldSymbol, initializations, cancellationToken),
+                        equivalenceKey: "Use member initializer"),
+                    context.Diagnostics);
+            }
+        }
+    }
+
+    private static async Task<Solution> EnforceFieldIsNotInitializedAsync(Document document, SyntaxNode declaration, CancellationToken cancellationToken)
+    {
+        SyntaxNode? root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root is not null && declaration is FieldDeclarationSyntax fieldDeclaration)
+        {
+            FieldDeclarationSyntax? newFieldDeclaration = CreateFieldDeclaration(fieldDeclaration);
+
+            if (newFieldDeclaration is not null)
+            {
+                return document.WithSyntaxRoot(root.ReplaceNode(fieldDeclaration, newFieldDeclaration)).Project.Solution;
+            }
+        }
+
+        return document.Project.Solution;
+    }
+
+    private static async Task<Solution> EnforceFieldDeclarationInitializationAsync(Document document, SyntaxNode declaration, CancellationToken cancellationToken)
+    {
+        SyntaxNode? root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root is not null && declaration is FieldDeclarationSyntax fieldDeclaration)
+        {
+            VariableDeclarationSyntax variableDeclaration = fieldDeclaration.Declaration;
+            VariableDeclaratorSyntax variable = variableDeclaration.Variables[0];
+            SemanticModel? semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            IFieldSymbol? fieldSymbol = semanticModel?.GetDeclaredSymbol(variable, cancellationToken) as IFieldSymbol;
+
+            if (fieldSymbol is not null)
+            {
+                EqualsValueClauseSyntax? newInitializer = GetInitializerForType(fieldSymbol.Type);
+
+                if (newInitializer != null)
+                {
+                    FieldDeclarationSyntax? fieldDeclarationWithInitializer = CreateFieldDeclaration(fieldDeclaration, newInitializer);
+
+                    if (fieldDeclarationWithInitializer is not null)
```

Comment:
Reduce nesting

Reply:
Finished V2 pass and added extra logic to handle another case I had not accounted for. Basically, all my tests used the default initializer, so they were passing. When the general rule is triggered, we want the same duplicate initializer to be moved up to the declaration (duplicating the node of course), instead of the default.